src/components/ListView/ListView.js

```diff
@@ -0,0 +1,180 @@
+import React, {
+  Component,
+  Fragment,
+} from 'react';
+import PropTypes from 'prop-types';
+import queryString from 'query-string';
+import {
+  get,
+  noop,
+} from 'lodash';
+
+import { IntlConsumer } from '@folio/stripes/core';
+
+import {
+  checkboxListShape,
+  trimSearchTerm,
+  withCheckboxList,
+} from '../../utils';
+import { SearchAndSort } from '../SearchAndSort';
+import { ViewContainer } from '../ViewContainer';
+import { listTemplate } from '../ListTemplate';
+import { ActionMenu } from '../ActionMenu';
+
+@withCheckboxList
+export class ListView extends Component {
+  static propTypes = {
+    resources: PropTypes.object.isRequired,
+    mutator: PropTypes.object.isRequired,
+    location: PropTypes.shape({ search: PropTypes.string.isRequired }).isRequired,
+    match: PropTypes.shape({ path: PropTypes.string.isRequired }).isRequired,
+    history: PropTypes.shape({ push: PropTypes.func.isRequired }).isRequired,
+    label: PropTypes.node.isRequired,
+    selectedEntity: PropTypes.object.isRequired,
+    checkboxList: checkboxListShape,
+    setList: PropTypes.func.isRequired,
+    showSingleResult: PropTypes.bool,
+    EntityView: PropTypes.func.isRequired,
+    EntityForm: PropTypes.func.isRequired,
+    INITIAL_RESULT_COUNT: PropTypes.number,
+    RESULT_COUNT_INCREMENT: PropTypes.number,
+    ENTITY_KEY: PropTypes.string,
+    actionMenuItems: PropTypes.arrayOf(PropTypes.string).isRequired,
+    visibleColumns: PropTypes.arrayOf(PropTypes.string).isRequired,
+    columnWidths: PropTypes.PropTypes.object.isRequired,
+    initialValues: PropTypes.object.isRequired,
+    renderHeaders: PropTypes.func.isRequired,
+  };
+
+  static defaultProps = {
+    showSingleResult: true,
+    checkboxList: {
```

Comment:
I don't think this default props are needed. `withCheckboxList` always passes this data

Reply:
Done